### DISCUSSION TITLE
Help generating ibl ktx file from jpg file

### DISCUSSION THREAD

**Discussion by paulorrfilho** [google/filament#3674] [Q&A] [answered]
How were the xxx_ibl.ktx files in the android samples generated?
I tried to create a ktx file using the Khronos Texture Tools (toktx) but it is not clear to me which settings I should use to work with Filament. I understand that requires a --cubemap type of file but toktx requires 6 input files, not sure what to do here.

**Comment by romainguy** [comment, marked as answer]
Those files are generates using the cmgen tool provided by Filament. This tool preprocesses environment maps to filter them as required by the engine. Our samples use a custom Gradle task to invoke cmgen with the appropriate flags.
